cql3/result_set.hh

```diff
@@ -193,6 +194,27 @@ public:
         }
     }
 
+    // visit_gently() is like visit(), except it may yield between rows so
+    // returns a future that will resolve when it's done. We only yield
+    // between rows, not between individual cells, which is a good compromise
+    // if we assume that individual rows are not too large.
+    template<typename Visitor>
+    requires ResultVisitor<Visitor>
+    future<> visit_gently(Visitor& visitor) const {
+        auto column_count = get_metadata().column_count();
+        for (auto& row : _rows) {
+            visitor.start_row();
+            for (auto i = 0u; i < column_count; i++) {
+                auto& cell = row[i];
+                visitor.accept_value(cell ? managed_bytes_view_opt(*cell) : managed_bytes_view_opt());
```

Comment:
It would be nicer to add to_managed_bytes_view_opt(const managed_bytes_opt&). But not required for this series.

Reply:
Indeed. But again, I just copied the existing code from visit().